test/python/transpiler/test_elide_permutations.py

```diff
@@ -207,6 +207,27 @@ def test_permutation_in_middle(self):
         res = self.swap_pass(qc)
         self.assertEqual(res, expected)
 
+    def test_partial_permutation_in_middle(self):
+        """Test with a permutation gate in the middle of a circuit,
+        with the permutation gate defined only on a subset of qubits.
+        """
+        qc = QuantumCircuit(5)
+        qc.cx(0, 1)
+        qc.append(PermutationGate([1, 2, 0]), [0, 2, 4])
+        qc.cx(2, 3)
+
+        # The permutation corresponding to the permutation gate maps
+        #   2 -> 0, 4 -> 2, 0 -> 4, and 1 -> 1, 3 -> 3.
+        # Instead of the permutation gate, we can relabel the qubits
+        #   0 -> 2, 1 -> 1, 2 -> 4, 3 -> 3, 4 -> 2.
+        # Hence cx(2, 3) becomes cx(4, 3).
+        expected = QuantumCircuit(5)
+        expected.cx(0, 1)
+        expected.cx(4, 3)
+
+        res = self.swap_pass(qc)
+        self.assertEqual(res, expected)
```

Comment:
Should we add some assertions on the permutation tracking too to make sure it's correct and reverses to the original circuit?

Reply:
This is a good idea. I have extended the test in [6791002](https://github.com/Qiskit/qiskit/pull/13186/commits/67910028e26eb8e5a013853cb31036ff9b27cca2). Do you think it's a good idea to add more "random circuits" tests, with swaps and permutations?

Reply:
Adding more tests wouldn't hurt. I think the one you added here provides sufficient coverage to prevent a regression here. But if you wanted to add more that would give us more coverage against potential future issues.